### PR TITLE
Support for Layer 2 networks with transaction fees on both layers

### PR DESCRIPTION
**Pull request by danjm**
This PR draws on work from https://github.com/MetaMask/metamask-extension/pull/12409 while simplifying that code.

The purpose of this PR is to modify our support for the Optimism chain so that fees will be displayed correctly after the upcoming hardfork. You can read about the fee changes here: https://community.optimism.io/docs/developers/l2/new-fees.html#for-frontend-and-wallet-developers

This PR supports those changes by showing the layer 1 fee, as retrieved from the Optimism gas oracle contract, on both the confirm screen and the transaction breakdown. It also disables editing of gas price, which would cause transactions on Optimism 2 to break.

Demo:

https://user-images.githubusercontent.com/7499938/141243970-95f0abef-e892-483f-b0ba-4b39b35aa504.mp4

### CODE REVIEW

File: ui/components/app/multilayer-fee-message/multi-layer-fee-message.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import React, { useContext, useState, useEffect } from 'react';
```
Comment (danjm) [inline ui/components/app/multilayer-fee-message/multi-layer-fee-message.js:1]: This is the file responsible for displaying the totals when on a "multilayer" network.

Screenshot from 2021-11-11 02-15-26

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 2]
```diff
@@ -473,14 +480,16 @@ export default class ConfirmTransactionBase extends Component {
...
                 detailText={
-                  <div className="confirm-page-container-content__currency-container">
-                    {renderHeartBeatIfNotInTest()}
-                    <UserPreferencedCurrencyDisplay
-                      type={SECONDARY}
-                      value={hexMinimumTransactionFee}
-                      hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
-                    />
-                  </div>
+                  !isMultiLayerFeeNetwork && (
```
Comment (danjm) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:483]: This does not change the code, only hides this component on `mutlilayerFeeNetwork`s. The multiline diff is just due to indentation changes.

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 3]
```diff
@@ -489,26 +498,30 @@ export default class ConfirmTransactionBase extends Component {
...
-                    {renderHeartBeatIfNotInTest()}
-                    <UserPreferencedCurrencyDisplay
-                      key="editGasSubTextFeeAmount"
-                      type={PRIMARY}
-                      value={hexMaximumTransactionFee}
-                      hideLabel={!useNativeCurrencyAsPrimaryCurrency}
-                    />
-                  </div>,
-                ])}
+                subText={
```
Comment (danjm) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:505]: This does not change the code, only hides this component on `mutlilayerFeeNetwork`s. The multiline diff is just due to indentation changes.

File: ui/pages/confirm-transaction-base/confirm-transaction-base.component.js [hunk 4]
```diff
@@ -537,19 +550,27 @@ export default class ConfirmTransactionBase extends Component {
...
-                renderTotalMaxAmount(),
-              ])}
-            />,
+            isMultiLayerFeeNetwork && (
+              <MultiLayerFeeMessage
+                transaction={txData}
+                layer2fee={hexMinimumTransactionFee}
+              />
+            ),
+            !isMultiLayerFeeNetwork && (
```
Comment (danjm) [inline ui/pages/confirm-transaction-base/confirm-transaction-base.component.js:559]: This does not change the code, only hides this component on `mutlilayerFeeNetwork`s. The multiline diff is just due to indentation changes.

**Review by jpuri — DISMISSED** [review]
Code changes look good, nice work 👍

File: ui/components/app/multilayer-fee-message/multi-layer-fee-message.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import TransactionDetailItem from '../transaction-detail-item/transaction-detail-item.component';
+import fetchEstimatedL1Fee from '../../../helpers/utils/optimism/fetchEstimatedL1Fee';
+import { I18nContext } from '../../../contexts/i18n';
+import { sumHexes } from '../../../helpers/utils/transactions.util';
+import {
+  toBigNumber,
+  toNormalizedDenomination,
+} from '../../../../shared/modules/conversion.utils';
+
+export default function MultilayerFeeMessage({ transaction, layer2fee }) {
```
Comment (mcmire) [inline ui/components/app/multilayer-fee-message/multi-layer-fee-message.js:13]: Ah, you decided to make a separate component for this. I like this implementation, seems easy to see what's going on.

**Review by mcmire — APPROVED** [review]
Had some comments, but overall this looks good to me. Thanks for taking this on for me :)

**Review by ryanml — APPROVED** [review]
looks good from my end! nice work

**Pull request merged**